common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -156,7 +158,9 @@ public static boolean execBinary(final UTF8String l, final UTF8String r) {
       return l.startsWith(r);
     }
     public static boolean execLowercase(final UTF8String l, final UTF8String r) {
-      return l.startsWithInLowerCase(r);
+      if (r.numBytes() == 0) return true;
+      if (l.numChars() < r.numChars()) return false;
```

Comment:
Same counterexample: Consider `l` being `İİİ` and `r` being `i̇i̇`. The answer should be true, but here we return false.